readthedocs/core/resolver.py

```diff
@@ -149,21 +149,29 @@ def resolve(self, project, protocol='http', filename='', private=None,
 
         # This duplication from resolve_domain is for performance purposes
         # in order to check whether a custom domain should be HTTPS
-        if custom_domain:
+        use_custom_domain = any([
+            (custom_domain and not require_https),
+            (custom_domain and require_https and custom_domain.https),
+        ])
```

Comment:
I don't think this logic is correct for .org (it is for .com). On .org, the canonical domain should be used regardless of HTTPS, correct? If the user specifies that their custom domain is canonical, all our links should use that I think.

Reply:
aye, this logic seems to be for the .com case, where we require https if you use a custom domain. perhaps this needs to be another method on the resolver, similar to _use_subdomain()

Reply:
In .org `require_https` will be `False` (the default). So this will enter because of the first condition. In that case, we will use the custom domain no matter what protocol we decide to use later in the next `if`.

On the other hand, if you are referring to the `canonical` attribute of the `Domain` object, we weren't considering it on this resolver at least. Actually, I'm not sure if it should be considered or not. I'm not sure how that attribute works.